### PR TITLE
WT-4839 Shuffle python tests and execute every Nth test

### PR DESCRIPTION
**Pull request by sulabhM**

### CODE REVIEW

File: test/suite/run.py [hunk 1]
```diff
@@ -117,6 +117,8 @@ def usage():
   -h      | --help               show this message\n\
   -j N    | --parallel N         run all tests in parallel using N processes\n\
   -l      | --long               run the entire test suite\n\
+  -o N    | --Nth N              randomly sort scenarios to be run, then\n\
```
Comment (luke-pearson) [inline test/suite/run.py]: Option here might be better as -N? so you'd do something like run.py -N 20? Could also be -r potentially for random.

**Comment by sulabhM** [comment]
@agorrod, have a look, please. Appreciated a suggestion for the name/letter for the option.

File: test/suite/run.py (hunk 1 above)
Comment (agorrod) [reply test/suite/run.py]: I agree re matching `-n` for the short version. How about leaving out a short option altogether and calling it `--random-sample`. I'd be happy with either of those choices.

**Comment by sulabhM** [comment]
@agorrod we already have -n for a dry run that lists all the scenarios. I put a change to have -r or --random-sample.
I think this looks good, I will checkin after the pull req tester passes.

**Pull request merged**